src/main/scanLogic/scanRunners/binaryRunner.ts

```diff
@@ -113,7 +114,7 @@ export abstract class BinaryRunner {
      * @param args  - the arguments for the command
      * @param executionLogDirectory - the directory to save the execution log in
      */
-    protected async executeBinary(checkCancel: () => void, args: string[], executionLogDirectory: string): Promise<void> {
+    protected async executeBinary(checkCancel: () => void, args: string[], executionLogDirectory?: string): Promise<void> {
```

Comment:
The older `.then` syntax should be replaced with the more modern and consistent `async/await` syntax. Please consider refactoring the code accordingly.